Extend the Llama-Nemotron-Nano-8B perf-integration-tests (cpp)

venkywonka opened the pull request:
## Description
Expand release-perf-regession-testing converage of `llama_v3.1_nemotron_nano_8b` to map to [NIM benchmarking configs](https://gitlab-master.nvidia.com/nim/benchmarking/-/blob/main/nim-llm/1.8/speed/nvidia-llama-3.1-nemotron-nano-8b-v1/speed_config.yaml?ref_type=heads) only for the cpp backend (as the PyT backend seems to be having issues that need further debugging).

---

### What’s inside
Cross product of the following:
```yaml
- model: llama_v3.1_nemotron_nano_8b
- runtime: bench
- backend: [trt]
- con: [1, 250] # concurrency
- input_output_len:
  - [5000, 500]
  - [500, 2000]
  - [1000, 1000]
  - [20000, 2000]
- quant: [none, fp8]
- dtype: [bfloat16]
- maxbs: [64]
```

> **Total:** 16 newly added cpp-backend perf tests`tests/integration/test_lists/qa/trt_llm_release_perf_test.yml`, removing previous cpp-backend perf tests, preserving the pyt-backend perf tests.

## 📊 Performance Benchmark Summary (subset)

As sanity check, below is perf overview of subset (8) of the 32 cases - that span the TRT flow on 1 H100.


| Input/Output Length | Concurrency | Precision | Request Throughput (req/s) | Output TPS (tokens/s) | P50 Latency (ms) |
|---------------------|-------------|-----------|-----------------------------|------------------------|------------------|
| 5000 / 500          | 1           | BF16      | 0.2312                      | 115.60                 | 4328.22          |
| 5000 / 500          | 1           | FP8       | 0.4163                      | 208.13                 | 2397.91          |
| 500 / 2000          | 1           | BF16      | 0.0616                      | 123.26                 | 16203.74         |
| 500 / 2000          | 1           | FP8       | 0.1051                      | 210.18                 | 9516.47          |
| 5000 / 500          | 250         | BF16      | 2.0712                      | 1035.62                | 117121.77        |
| 5000 / 500          | 250         | FP8       | 3.6120                      | 1805.98                | 67227.43         |
| 500 / 2000          | 250         | BF16      | 0.8633                      | 1726.57                | 278001.38        |
| 500 / 2000          | 250         | FP8       | 1.4663                      | 2932.65                | 163683.41        |

### 🧵 Observations

- **FP8 consistently improves** both request throughput and token throughput across all configurations, with up to **2× higher output TPS** in high concurrency scenarios.
- **Latency** (P50) significantly reduces with FP8, especially under low concurrency, showing strong inference-time gains.
- **Concurrency scaling**: Increasing concurrency to 250 leads to significantly higher overall throughput, though with expected increases in tail latencies.

### 📂 Dataset Details

Each configuration used synthetic datasets generated with consistent parameters (`512` sequences per run) using the script in `TensorRT-LLM/benchmarks/cpp/prepare_dataset.py`.

venkywonka committed:
add ll-nm-nano tests that map to nim requirements

Signed-off-by: Venky <23023424+venkywonka@users.noreply.github.com>

ruodil commented on tests/integration/test_lists/qa/trt_llm_release_perf_test.yml:
```diff
@@ -26,12 +26,40 @@ trt_llm_release_perf_test:
+  - perf/test_perf.py::test_perf[llama_v3.1_nemotron_nano_8b-bench-pytorch-bfloat16-maxbs:64-input_output_len:20000,2000-con:1]
+  - perf/test_perf.py::test_perf[llama_v3.1_nemotron_nano_8b-bench-pytorch-bfloat16-maxbs:64-input_output_len:20000,2000-quant:fp8-con:1]
+  - perf/test_perf.py::test_perf[llama_v3.1_nemotron_nano_8b-bench-pytorch-bfloat16-maxbs:64-input_output_len:5000,500-con:250]
+  - perf/test_perf.py::test_perf[llama_v3.1_nemotron_nano_8b-bench-pytorch-bfloat16-maxbs:64-input_output_len:5000,500-quant:fp8-con:250]
```
when using pytorch backend, it will not do quantization even if specifying 'quant:fp8' in test case, and pytorch backend will use original datatype to forward, please use a fp8 model instead, thank!

venkywonka replied:
I see, i believe we don't have a super fp8 pre-quantized yet - so for now i shall just remove them.

ruodil commented on tests/integration/test_lists/qa/trt_llm_release_perf_test.yml:
```diff
@@ -26,12 +26,40 @@ trt_llm_release_perf_test:
+  - perf/test_perf.py::test_perf[llama_v3.1_nemotron_nano_8b-bench-pytorch-bfloat16-maxbs:64-input_output_len:1000,1000-con:250]
+  - perf/test_perf.py::test_perf[llama_v3.1_nemotron_nano_8b-bench-pytorch-bfloat16-maxbs:64-input_output_len:1000,1000-quant:fp8-con:250]
+  - perf/test_perf.py::test_perf[llama_v3.1_nemotron_nano_8b-bench-pytorch-bfloat16-maxbs:64-input_output_len:20000,2000-con:250]
+  - perf/test_perf.py::test_perf[llama_v3.1_nemotron_nano_8b-bench-pytorch-bfloat16-maxbs:64-input_output_len:20000,2000-quant:fp8-con:250]
```
and there are many cases with large isl/osl, I'm not sure how much time it will cost, and do we need to put these nemo cases in an extra yml test list? cc @LarryXFly

LarryXFly replied:
Let's first expand the test coverage for this model and move some of the tests to a separate test list for NIM.

venkywonka committed:
prune some pytorch cases (fp8)

Signed-off-by: Venky <23023424+venkywonka@users.noreply.github.com>

venkywonka commented:
I'm locally seeing issue with the above configurations when using pyt backend, hence limiting this PR to only cpp backend. Don't want to check-in tests that break.

venkywonka committed:
removing pyt backend test changes

- When validating the pytorch tests with the isl/osl/conc/quant settings (that is done for cpp backend too), seeing hangs that need further debugging.
- Therefore don't want to block this PR, hence removing them.
- Seeing

Signed-off-by: Venky <23023424+venkywonka@users.noreply.github.com>

venkywonka commented:
/bot run

tensorrt-cicd commented:
[PR_Github #4892](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4892/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #4892](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4892/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3544](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3544/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #4901](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4901/) [ run ] triggered by Bot

venkywonka commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #4904](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4904/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #4901](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4901/) [ run ] completed with state `ABORTED`

tensorrt-cicd commented:
[PR_Github #4904](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4904/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3556](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3556/) completed with status: 'FAILURE'

venkywonka commented:
/bot run

tensorrt-cicd commented:
[PR_Github #5348](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5348/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #5348](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5348/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3901](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3901/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #5531](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5531/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #5531](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5531/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #4032](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/4032/) completed with status: 'FAILURE'

venkywonka commented:
/bot run

tensorrt-cicd commented:
[PR_Github #5545](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5545/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #5545](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5545/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #4046](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/4046/) completed with status: 'FAILURE'

chzblych commented:
/bot skip --comment "Irrelevant TOT failures"

tensorrt-cicd commented:
[PR_Github #5573](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/5573/) [ skip ] triggered by Bot
